crates/aptos-crypto/src/bls12381/bls12381_keys.rs

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module provides APIs for private keys and public keys used in BLS multi-signatures
+//! implemented on top of BLS12-381 elliptic curves (https://github.com/supranational/blst).
+//!
+//! The `PublicKey` struct is used to represent both the public key of an individual signer
+//! as well as the aggregate public key of several signers. Before passing such a struct as an
+//! argument the caller should always ensure it has verified its proof-of-possession via
+//! `ProofOfPossession::verify`.
+//!
+//! The `PublicKey::aggregate` API assumes the caller has already verified
+//! proofs-of-possession for all the given public keys and therefore all public keys are valid group
+//! elements.
+//!
+//! In general, with the exception of `ProofOfPossession::verify` no library function should
+//! be given a public key as argument without first verifying that public key's PoP. Note that
+//! for aggregate public keys obtained via `PublicKey::aggregate` there is no PoP to verify, but
+//! the security assumption will be that all public keys given as input to this function have had
+//! their PoPs verified.
+
+use std::convert::TryFrom;
+use anyhow::{anyhow, Result};
+use serde::Serialize;
+use aptos_crypto_derive::{DeserializeKey, SerializeKey, SilentDebug, SilentDisplay};
+use crate::{bls12381, traits, CryptoMaterialError, Length, signing_message, Uniform, ValidCryptoMaterial, ValidCryptoMaterialStringExt, VerifyingKey};
+use crate::bls12381::DST_BLS_MULTISIG_IN_G2_WITH_POP;
+use crate::hash::CryptoHash;
+
+#[derive(Clone, Eq, PartialEq, SerializeKey, DeserializeKey)]
+/// A BLS12381 public key
+pub struct PublicKey {
+    pub(crate) pubkey: blst::min_pk::PublicKey,
+
+    // NOTE: We do not keep the PoP here in order to minimize the size of this struct, since we'll be
+    // storing these PKs in the root of the Merkle accumulator (according to Zekun).
```

Comment:
Suggest rewrite this as: "NOTE: In order to minimize the size of this struct, we do not keep the PoP here. Moreover, these PKs are stored in the root of the Merkle accumulator." 
- I removed "according to Zekun" because I thought this kind of note is an internal note. I have no strong opinions on it though.